### PR TITLE
Implement nats in kuber, add more tests

### PR DESCRIPTION
**Pull request by ajax-yurkovskyi-t**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -11,14 +11,29 @@ services:
       - db-data:/data/db
 
   app:
-    build:
-      context: .
+    image: iot-management-device
```
Comment (ajax-gnatiuk-s) [inline docker-compose.yaml:14]: who builds this image?

File: gateway/src/main/kotlin/com/example/gateway/mapper/EnumMapper.kt [hunk 2]
```diff
@@ -3,17 +3,16 @@ package com.example.gateway.mapper
...
-        ValueMapping(source = "STATUS_TYPE_UNSPECIFIED", target = "OFFLINE"),
-        ValueMapping(source = "UNRECOGNIZED", target = "OFFLINE"),
-    )
-    fun mapStatusType(statusType: Device.StatusType): DeviceStatusType
+    fun mapStatusType(statusType: Device.StatusType): DeviceStatusType {
+        return when (statusType) {
+            Device.StatusType.STATUS_TYPE_UNSPECIFIED -> DeviceStatusType.OFFLINE
+            Device.StatusType.STATUS_TYPE_ONLINE -> DeviceStatusType.ONLINE
+            Device.StatusType.STATUS_TYPE_OFFLINE -> DeviceStatusType.OFFLINE
+            Device.StatusType.UNRECOGNIZED -> DeviceStatusType.OFFLINE
```
Comment (ajax-gnatiuk-s) [inline gateway/src/main/kotlin/com/example/gateway/mapper/EnumMapper.kt:15]:
```suggestion
            Device.StatusType.STATUS_TYPE_ONLINE -> DeviceStatusType.ONLINE
            Device.StatusType.STATUS_TYPE_OFFLINE, Device.StatusType.UNRECOGNIZED, Device.StatusType.STATUS_TYPE_UNSPECIFIED -> DeviceStatusType.OFFLINE
```

File: gateway/src/test/kotlin/com/example/gateway/mapper/EnumMapperTest.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+
+class EnumMapperTest {
+    private val enumMapper = EnumMapperImpl()
+
+    @ParameterizedTest
+    @MethodSource("provideStatusMappingCases")
```
Comment (ajax-gnatiuk-s) [inline gateway/src/test/kotlin/com/example/gateway/mapper/EnumMapperTest.kt:14]:
```suggestion
    @CsvSource(
        textBlock = """
        STATUS_TYPE_UNSPECIFIED, OFFLINE
        STATUS_TYPE_ONLINE, ONLINE
        STATUS_TYPE_OFFLINE, OFFLINE
        UNRECOGNIZED, OFFLINE"""
    )
```

File: iot-management-device/src/main/kotlin/com/example/iotmanagementdevice/mapper/EnumMapper.kt [hunk 4]
```diff
@@ -3,21 +3,23 @@ package com.example.iotmanagementdevice.mapper
...
-    @ValueMapping(source = "STATUS_TYPE_OFFLINE", target = "OFFLINE")
-    @ValueMapping(source = "STATUS_TYPE_UNSPECIFIED", target = "OFFLINE")
-    @ValueMapping(source = "UNRECOGNIZED", target = "OFFLINE")
-    fun mapStatusType(statusType: Device.StatusType): DeviceStatusType
+    fun mapStatusType(statusType: Device.StatusType): DeviceStatusType {
+        return when (statusType) {
+            Device.StatusType.STATUS_TYPE_UNSPECIFIED -> DeviceStatusType.OFFLINE
+            Device.StatusType.STATUS_TYPE_ONLINE -> DeviceStatusType.ONLINE
+            Device.StatusType.STATUS_TYPE_OFFLINE -> DeviceStatusType.OFFLINE
+            Device.StatusType.UNRECOGNIZED -> DeviceStatusType.OFFLINE
```
Comment (ajax-gnatiuk-s) [inline iot-management-device/src/main/kotlin/com/example/iotmanagementdevice/mapper/EnumMapper.kt:22]:
```suggestion
            Device.StatusType.STATUS_TYPE_ONLINE -> DeviceStatusType.ONLINE
            Device.StatusType.STATUS_TYPE_UNSPECIFIED, Device.StatusType.STATUS_TYPE_OFFLINE, Device.StatusType.UNRECOGNIZED -> DeviceStatusType.OFFLINE
```

File: gateway/src/test/kotlin/com/example/gateway/mapper/EnumMapperTest.kt (hunk 3 above)
Comment (ajax-yurkovskyi-t) [reply gateway/src/test/kotlin/com/example/gateway/mapper/EnumMapperTest.kt]: changed to @CsvSource

File: docker-compose.yaml (hunk 1 above)
Comment (ajax-yurkovskyi-t) [reply docker-compose.yaml:14]: me manually, fixed it

**Review by ajax-samondros-b — APPROVED** [review]

**Pull request merged**